python/pythonlab/neighborhood/neighborhood/support/grid.py

```diff
@@ -0,0 +1,28 @@
+from .neighborhood_runtime_exception import NeighborhoodRuntimeException
+from .exception_keys import ExceptionKeys
+from .grid_square import GridSquare
+
+class Grid:
+    def __init__(self, squares: list[list[GridSquare]]):
+        self.grid = squares
+        self.height = len(squares)
+        self.width = len(squares[0])
+
+    def print_grid(self):
+        for y in range(self.height):
+            squares = [self.grid[x][y].get_printable_description() for x in range(self.width)]
+            print(",".join(squares))
+
+    def valid_location(self, x: int, y: int) -> bool:
+        # A coordinate cannot be moved into if it is out of range or if the tile is not passable
+        return 0 <= x < self.width and 0 <= y < self.height and self.grid[x][y].is_passable()
+
+    def get_square(self, x: int, y: int) -> GridSquare:
+        if self.valid_location(x, y):
+            return self.grid[x][y]
+        else:
+            raise NeighborhoodRuntimeException(ExceptionKeys.GET_SQUARE_FAILED)
+
+    def get_size(self) -> int:
```

Comment:
If `grid` is always a square, then could we just have instance variable `length` instead of `width` and `height`?

Reply:
Good point. We always make it a square on the frontend, but the api doesn't block a non-square grid. I'm thinking we throw an exception in init if `squares` is not a square, what do you think?

Reply:
Sure - sounds good!